### PR TITLE
AP: Always sign HTTP requests

### PR DESCRIPTION
**Pull request by annando**
See here: https://forum.friendi.ca/display/d8230ed6-515f-2083-ca77-075356461160

Mastodon can be configured to only accept signed HTTP requests - so we now sign them. SInce we don't always work in a user context we simply use the administrator or the first found user.

### CODE REVIEW

File: src/Model/User.php [hunk 1]
```diff
@@ -185,6 +185,29 @@ public static function getByEmail($email, array $fields = [])
...
 
+	/**
+	 * Fetch the user array of the administrator. The first one if there are several.
+	 *
+	 * @param array $fields
+	 * @return array user
+	 */
+	public static function getFirstAdmin(array $fields = [])
+	{
+		$condition = [];
```
Comment (MrPetovan) [inline src/Model/User.php:196]: `$condition` is never used in this method.

Comment (annando) [reply src/Model/User.php]: Not only this - but the functionality had been weird. That's what happens when you perform "cut&paste" at 5 in the morning :-)

**Comment by annando** [comment]
@MrPetovan is there something left to do here?

**Comment by MrPetovan** [comment]
I was waiting for the test results and then life happened.

**Pull request merged**